src/errors.rs

```diff
@@ -12,6 +14,14 @@ error_type! {
     pub enum SpaceapiServerError {
         Redis(RedisError) {
             cause;
-        }
+        },
+        IoError(io::Error) {
+            cause;
+        },
+        Message(Cow<'static, str>) {
+            desc (e) &**e;
+            from (s: &'static str) s.into();
+            from (s: String) s.into();
+        },
```

Comment:
> A semantically relevant name, e.g.  ArgumentParsing .

Ok but otherwise it would stay the same? Just rename to `ArgumentParsing`.
> Btw, where does the  IoError  come from? Maybe we could wrap that too?

What do you mean? It comes from `io::Error` isn't that obvious? And we wrap it inside the `SpaceapiServerError`. And it's caused by
```Rust
let mut socket_addr_iter = try!(socket_addr.to_socket_addrs());
```